src/noise.cpp

```diff
@@ -583,8 +574,7 @@ void Noise::gradientMap3D(
 	u32 nlx, nly, nlz;
 	s32 x0, y0, z0;
 
-	Interp3dFxn interpolate = (np.flags & NOISE_FLAG_EASED) ?
```

Comment:
consider a macro
```cpp
#define interpolate(...) eased ? triLinearInterpolation(__VA_ARGS__) : triLinearInterpolationNoEase(__VA_ARGS__)
```

Reply:
How about combining the methods and using an `eased` flag? I added this, and verified it has the same improved performance.

Reply:
Sure, just add something like `// The compiler is expected to optimize out this if() when inlining` or similar in the right place

Reply:
Added a comment. Honestly not sure if the compiler will optimize it out (unless `eased` is specified as literal true/false at compile time). Even so, a branch is much faster than a function call through a pointer. CPU branch predictors are very good, especially when the branch is exactly the same repeatedly. The CPU's pipeline will make it as cheap as a NOP.

The compiler may also choose to hoist the branch up one or two levels of the loop. Inlining makes a lot of things possible.